tests/catch/unit/memory/MemUtils.hh

```diff
@@ -452,15 +618,21 @@ static inline void verifyData(T* aPtr, size_t value, MultiDData& data, allocType
 
 // function used to abstract the test
 template <typename T, typename F, typename... fArgs>
-static inline void doMemsetTest(F func, fArgs... funcArgs) {
+static inline void doMemTest(F func, fArgs... funcArgs) {
+  bool async = GENERATE(true, false);
   enum StreamType { NULLSTR, CREATEDSTR };
-  auto streamType = GENERATE(NULLSTR, CREATEDSTR);
+  StreamType streamType;
+  if(async) {
+   streamType = GENERATE(NULLSTR, CREATEDSTR);
+  } else {
+    streamType = NULLSTR;
+  }
   hipStream_t stream{nullptr};
 
   if (streamType == CREATEDSTR) HIP_CHECK(hipStreamCreate(&stream));
-  func(stream, funcArgs...);
+  func(stream, async, funcArgs...);
   if (streamType == CREATEDSTR) HIP_CHECK(hipStreamDestroy(stream));
 }
```

Comment:
This is getting too complex, can I suggest this:
```cpp
 SECTION("synchronous") { func(nullptr, false, funcArgs...); }
  SECTION("asynchronous - null stream") { func(nullptr, true, funcArgs...); }
  SECTION("asynchronous - created stream") {
    hipStream_t stream{};
    HIP_CHECK(hipStreamCreate(&stream));
    func(stream, true, funcArgs...);
    HIP_CHECK(hipStreamDestroy(stream));
  }
```

Reply:
Suggestion is much cleaner, thank you